torchx/schedulers/ray/ray_driver.py

```diff
@@ -77,77 +102,189 @@ def load_actor_json(filename: str) -> List[RayActor]:
 
-    return cmd_actors
+        # get the actor id of the created actor
+        actor_id = actor._actor_id.hex()
+        # launch a task to check if the actor is scheduled
+        self.active_tasks.append(actor.schedule.remote(actor_id))
+        # save the actor info for recovering from node failures
+        self.actor_info_of_id[actor_id] = ActorInfo(
+            actor=actor,
+            pg=pg,
+            replica=replica,
+        )
+
+    def place_command_actors(self) -> None:
+        """Creating all command actors in all placement groups"""
+        # find the placement group index for a replica(actor's specification)
+        pg_ix_of_replica: List[int] = [
+            max(0, i - self.min_nnodes + 1)  # pyre-ignore
+            for i in range(len(self.replicas))
+        ]
+        # create the actors
+        for i in range(len(self.replicas)):
+            pg_ix = pg_ix_of_replica[i]
+            pg = self.placement_groups[pg_ix]  # find the created placement group
+            replica = self.replicas[i]
+            self.create_and_schedule_actor(pg, replica)
+
+    def _step(self) -> bool:
+        """Handling command actor's return"""
+        result: RayResult  # execution result
+        _logger.info(f"running ray.wait on {self.active_tasks}")
+        # ray.wait is partial waiting
+        # pyre-fixme[16]: Module `worker` has no attribute `wait`.
+        completed_tasks, self.active_tasks = ray.wait(self.active_tasks)
+        # If a failure occurs the ObjectRef will be marked as completed.
+        # Calling ray.get will expose the failure as a RayActorError.
+        for object_ref in completed_tasks:
+            result = ray.get(object_ref)  # pyre-ignore
+            if isinstance(result, CommandActorScheduled):
+                if self.need_more_actors:
+                    actor = self.actor_info_of_id[result.id].actor
+                    if self.master_node_id is None:
+                        # make this actor be the master node
+                        self.master_node_id = result.id
+                        self.rank_0_address, self.rank_0_port = ray.get(
+                            actor.get_actor_address_and_port.remote()  # pyre-ignore
+                        )
+                        self.active_tasks.append(
+                            actor.exec_module.remote(  # pyre-ignore
+                                "localhost", 0, result.id
+                            )
+                        )
+                    else:
+                        self.active_tasks.append(
+                            actor.exec_module.remote(
+                                self.rank_0_address, self.rank_0_port, result.id
+                            )
+                        )
+                    self.command_actors_count += 1
+            elif isinstance(result, TaskCompleted):
+                self.need_more_actors = False  # don't need more actors
```

Comment:
maybe should name this terminating? and then we can do `if not self.terminating` above